### DISCUSSION TITLE
Fade in audio

### DISCUSSION THREAD

**Discussion by tobx** [pdeljanov/Symphonia#225] [General]
Hello,

I have some experience with rust, but not all that much. It is difficult for me to grasp all of this, so I hope for some help here. My goal is to play and slowly fade in audio (smoothly, sample by sample) for a configurable duration.

So, using `symphonia-play` as a starting point, is this here the recommended point to hop in and do the fade in calculations:
https://github.com/pdeljanov/Symphonia/blob/10f4fa0d9d955f19fca6236523d2cf58e3c67a47/symphonia-play/src/main.rs#L375C34-L375C34

But how can I get the actual samples from `decoded` to do the math with?

**Comment by aryalaadi** [comment]
```rs
let packet = match format.next_packet() {
 Ok(p) => p,
 Err(_) => break,
 };
let decoded = match decoder.decode(&packet) {
 Ok(d) => d,
 Err(_) => continue,
};              
let spec = *decoded.spec();
let duration = decoded.capacity() as u64;
let mut buf = SampleBuffer::<f32>::new(duration, spec);
buf.copy_interleaved_ref(decoded);
for sample in buf.samples() {
  // your math goes in here
}
```

This is probably the most simple way for you to do math on the samples, if you wanted to know exactly where to add the math in `symphonia-play` it would be in `output.rs`
